app/src/ui/lib/configure-git-user.tsx

```diff
@@ -42,27 +45,64 @@ export class ConfigureGitUser extends React.Component<
   public constructor(props: IConfigureGitUserProps) {
     super(props)
 
-    this.state = { name: '', email: '', avatarURL: null }
+    this.state = {
+      globalUserName: null,
+      globalUserEmail: null,
+      name: '',
+      email: '',
+      avatarURL: null,
+    }
   }
 
   public async componentWillMount() {
-    let name = await getGlobalConfigValue('user.name')
-    let email = await getGlobalConfigValue('user.email')
+    const [globalUserName, globalUserEmail] = await Promise.all([
+      getGlobalConfigValue('user.name'),
+      getGlobalConfigValue('user.email'),
+    ])
+
+    this.setState(prevState => ({
+      globalUserName,
+      globalUserEmail,
+      name: prevState.name.length === 0 ? globalUserName || '' : prevState.name,
+      email:
+        prevState.email.length === 0 ? globalUserEmail || '' : prevState.email,
+    }))
+  }
 
-    const user = this.props.accounts[0]
-    if ((!name || !name.length) && user) {
-      name = user.name && user.name.length ? user.name : user.login
+  public componentDidMount() {
+    if (this.props.accounts.length > 0) {
+      this.setDefaultValuesFromAccount(this.props.accounts[0])
```

Comment:
I'm curious about this accounts array... don't have time at the moment to investigate it, but what I'm wondering is how we can be so sure that the first element is what we want? Same for below in `componentDidUpdate`

Reply:
This is old logic but the thought is that unless you're fiddling with `localStorage` there's no way you can run the welcome wizard twice and the welcome flow is designed such that when you're done you're signed in to one account. It's technically possible to sign in to two accounts in the welcome flow but chances that you'd end up doing that is pretty slim so picking the first account seemed safe enough at the time and we haven't had any complaints that I know of regarding that logic.